core/logstore/logstore.go

```diff
@@ -61,12 +64,18 @@ type ThreadMetadata interface {
 	// PutInt64 stores an int value under key.
 	PutInt64(t thread.ID, key string, val int64) error
 
-	// GetString retrieves an int value under key.
+	// GetString retrieves a string value under key.
 	GetString(t thread.ID, key string) (*string, error)
 
 	// PutString stores a string value under key.
 	PutString(t thread.ID, key string, val string) error
 
+	// GetBool retrieves a boolean value under key.
+	GetBool(t thread.ID, key string) (*bool, error)
```

Comment:
Adding here as a useful 'flag' API, but a more dedicated/bespoke API could be used instead to create the 'direct' flag elsewhere?

Reply:
SGTM!